Vendor dependencies, update k8s to 1.16.3

tfussell opened the pull request:
Towards giantswarm/giantswarm#7675 and https://github.com/giantswarm/roadmap/issues/19

Updates kube-state-metrics to upstream version 1.9.0 (1.8 isn't compatible with 1.16) and replaces use of `e2esetup/k8s` package with `k8sclient`.

tfussell commented:
e2e failed but CircleCI considered it to have passed for some reason. Not sure what's going on. 
```
--- FAIL: TestHelm (5.58s)
    basic_test.go:13: [{/go/src/github.com/giantswarm/kube-state-metrics-app/vendor/github.com/giantswarm/e2etests/basicapp/basicapp.go:95: } {/go/src/github.com/giantswarm/kube-state-metrics-app/vendor/github.com/giantswarm/e2etests/basicapp/legacyresource/resource.go:105: } {/go/src/github.com/giantswarm/kube-state-metrics-app/vendor/github.com/giantswarm/helmclient/helmclient.go:330: } {/go/src/github.com/giantswarm/kube-state-metrics-app/vendor/github.com/giantswarm/helmclient/helmclient.go:375: } {/go/src/github.com/giantswarm/kube-state-metrics-app/vendor/github.com/giantswarm/backoff/retry.go:23: } {/go/src/github.com/giantswarm/kube-state-metrics-app/vendor/github.com/giantswarm/helmclient/helmclient.go:350: } {rpc error: code = Unknown desc = a release named kube-state-metrics already exists.
        Run: helm ls --all kube-state-metrics; to check the status of the release
        Or run: helm del --purge kube-state-metrics; to delete it}]
FAIL
ok  	github.com/giantswarm/kube-state-metrics-app/integration/test/basic	20.121s
```

sslavic commented on Gopkg.lock:
```diff
@@ -1354,112 +1374,63 @@
+  ]
+  pruneopts = "UT"
+  revision = "190eafce4a149d58ed93c5ca2e546244da4837d4"
+  version = "v0.16.4"
```
Please add override for kubectl transitive dependency and set it to v0.16.3 so it's aligned with the rest of the k8s dependencies

tfussell replied:
Fixed.

sslavic commented on Gopkg.lock:
```diff
@@ -1354,112 +1374,63 @@
-  revision = "67d2fcf276fcd9cf743ad4be9a9ef5828adc082f"
-  version = "v1.15.4"
+  revision = "224be7bdce5a9dd0c2fd0d46b83865648e2fe0ba"
+  version = "v1.16.4"
```
Please adjust override for k8s.io/kubernetes transitive dependency and set it to `=v0.16.3` so it's aligned with the rest of the k8s dependencies

tfussell replied:
I didn't realize that the version specification without qualifiers was treated like >=. This is fixed now, thanks.

tfussell commented:
I removed `github.com/Azure/go-autorest` as it didn't seem to be being used. I moved `k8s.io/api` from constraint to override as it's not a direct dependency.

Gacko commented:
Please leave releasing to the code owners, thank you!

QuantumEnigmaa commented:
Sorry, I just wanted to have a first release so that I could try it on a testing installation :bow:

Gacko commented:
You can always use the test catalog for that. But right now I do not feel like the app is in a state it's safe to be released. I'd still like to do some chores and improvements, but leave them to us.
